server/src/main/java/org/apache/druid/client/JsonParserIterator.java

```diff
@@ -127,7 +139,7 @@ private void init()
           throw new QueryInterruptedException(
               new ResourceLimitExceededException(
                   "query[%s] url[%s] timed out or max bytes limit reached.",
-                  query.getId(),
+                  query != null ? query.getId() : null,
```

Comment:
thanks for catching and fixing potential NPE here

Reply:
Please annotate `query` with `@Nullable`.